### DISCUSSION TITLE
Several Questions & Ideas about different plugins

### DISCUSSION THREAD

**Discussion by Anmol-Baranwal** [lowlighter/metrics#1397] [General]
## [Notable Contributions](https://github.com/lowlighter/metrics/blob/master/source/plugins/notable/README.md)

> I want to know more about `plugin_notable_indepth`

![why 1 notable contribution is shining](https://user-images.githubusercontent.com/74038190/223761596-97acfda2-adab-432e-9bec-348cdecdf3f8.png)

- Q1: Why is the color of one contribution different from the others?
- Q2: What are the three icons? (PR is understandable)
- Idea: The color of icons should be different since it's not visible in dark mode. Accessibility is important for open-source.

## [Achievement](https://github.com/lowlighter/metrics/blob/master/source/plugins/achievements/README.md)

![image](https://user-images.githubusercontent.com/74038190/223762469-4d2155b6-5119-42ec-9f96-d1c3ea7905aa.png)

- Q: Where can I read more about each of the achievements?

## [Leetcode](https://github.com/lowlighter/metrics/blob/master/source/plugins/leetcode/README.md)

> The plugin

![image](https://user-images.githubusercontent.com/74038190/223764422-cf574dca-b3c7-4a31-a3ee-ee7280f9ce0d.png)

> Leetcode Profile

![image](https://user-images.githubusercontent.com/74038190/223764463-d9bf44c6-9131-49bd-b872-f23d4f2a7c83.png)

>As you can see, it just retrieves the higher number of skills in sorting order. 

- Idea: There could be different limit options to choose skills from each category: `Advanced`, `Intermediate`, and `Fundamental`. 
I understand that the color on the plugin stats indicates the level of skill, however in some cases, intermediate skill may be more significant to exhibit than advanced skill.

**Comment by lowlighter** [comment]
Hi 👋 !

### Notable contributions

A1. The different colors is supposed to highlight the repo where you made significant contributions. The color scheme is similar to the one used by the achievement plugin, with the exception of the "red" one being when you're a maintained (i.e. have push access) to the repository (and the token was able to know you have rw rights). The threshold are 20%, 10% and 5% of overall commits:

https://github.com/lowlighter/metrics/blob/ba5d3c4931bd522e82a17691ff96eca8578a80b5/source/templates/classic/partials/notable.ejs#L20

A2. The three icons are commits number, "relative stars" number (basically overall contribution percentage times the stars of repo) and number of pull requests. These are flat numbers while the gauge behind supposed to display the percentage value

As for accessibility it's a bit tricky, because these are supposed to be visible on both light/dark mode which is why gray is used so it's an in-between that, but I conceed it's not that visible. But using colors that contrast too much would make it weird so I have not found a satisfying color yet. Maybe a chromatic expert could help us more on this 🙂 

### Achievements

A. So the achievements descriptions are only visible when using [`plugin_achievements_display: detailed`](https://github.com/lowlighter/metrics/tree/master/source/plugins/achievements#plugin_achievements_display), but if you don't mind reading code you can see the descriptions and threshold in this file [`source/plugins/achievements/list/users.mjs`](https://github.com/lowlighter/metrics/blob/master/source/plugins/achievements/list/users.mjs). The threshold are voluntarly high to make them revelant and get a quick overview of what a user uses primarly GitHub for

### Leetcode

Adding filter/limit options is a good idea, I'll open an issue for this 👍 


Thanks a lot for these suggestions 😄  !
